pkg/controller/plan/adapter/openstack/builder_test.go

```diff
@@ -0,0 +1,60 @@
+package openstack
+
+import (
+	"testing"
+)
+
+func TestGetPreferenceOs(t *testing.T) {
+	tests := []struct {
+		name       string
+		os         string
+		version    string
+		distro     string
+		expectedOs string
+	}{
+		{
+			name:       "RHEL9",
+			os:         RHEL,
+			version:    "9",
+			distro:     RHEL,
+			expectedOs: "rhel.9",
+		},
+		{
+			name:       "Windows2022",
+			os:         Windows,
+			version:    "2022",
+			distro:     Windows,
+			expectedOs: "windows.2k22",
+		},
+		{
+			name:       "Windows11",
+			os:         Windows,
+			version:    "11",
+			distro:     Windows,
+			expectedOs: "windows.11",
+		},
+		{
+			name:       "Centos stream 9",
+			os:         CentOS,
+			version:    "9",
+			distro:     CentOS,
+			expectedOs: "centos.stream9",
+		},
+		{
+			name:       "Ubuntu 22.04.3",
+			os:         Ubuntu,
+			version:    "22.04.3",
+			distro:     Ubuntu,
+			expectedOs: "ubuntu",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			actualOs := getPreferenceOs(tt.os, tt.version, tt.distro)
+			if actualOs != tt.expectedOs {
+				t.Errorf("got (%s), want (%s)", actualOs, tt.expectedOs)
+			}
+		})
+	}
```

Comment:
how about using a Table like we do [here](https://github.com/kubev2v/forklift/blob/main/pkg/controller/provider/container/vsphere/collector_test.go#L29-L36)?

Reply:
+1, also this is using plain testing without Ginkgo/Gomega atm.